products/views.py

```diff
@@ -25,4 +26,74 @@ def get(self, request, product_id):
             "category"         : product.sub_category.category.category,
         }
 
-        return JsonResponse(response, status=200)
\ No newline at end of file
+        return JsonResponse(response, status=200)
+
+class ProductListView(View):
+    def get(self, request):
+        group               = request.GET.get('group', None)
+        sub_categories      = request.GET.getlist('sub-category', None)
+        colors              = request.GET.getlist('color', None)
+        sizes               = request.GET.getlist('size', None)
+        price_ranges        = request.GET.getlist('price', None)
+        sort_by             = request.GET.get('sort', None)
+
+        group_filter        = Q()
+        sub_category_filter = Q()
+        color_filter        = Q()
+        size_filter         = Q()
```

Comment:
Q을 활용하신 점 잘 하셨습니다. 
제가 조금 조언을 드리자면, 보통 Q을 사용할 때는 빈 Q을 1번만 선언하고 선언한 Q에 계속 조건을 쌓아가는 형식을 많이 사용합니다.
```suggestion
       q = Q()
       
        if group:
            if group == 'men':
                q &= Q(group='남자')
            else:
                q &= Q(group='여자')
```

Reply:
Q 객체 하나로 필터 쿼리를 쌓을 수 있도록 코드를 수정해보겠습니다